### PR TITLE
tests: for gRPC replays, establish a local listener to connect to

### LINKED ISSUE #793: runtimeconfigurator: test hangs when run locally in replay mode with no internet connection
### To Reproduce

Turn off wifi, unplug ethernet.
```
cd runtimevar/runtimeconfigurator
go test -v
```

At this point it hangs. Hit ctrl-\ to get a stack trace. That shows that it hangs in `grpc.DialContext`.

## Expected behavior

The test should pass.

### Version

a36581c5b990d15cef1835c94f3cbf11f5f78cce

### PR DESCRIPTION
**Pull request by vangent**
Per suggestion from [here](https://github.com/GoogleCloudPlatform/google-cloud-go/issues/1239#issuecomment-443300398).

After this PR, tests should work without a network connection.

Fixes #793.

### CODE REVIEW

**Comment by jba** [comment]
I assigned myself to that cloud.google.com/go issue to see if rpcreplay can do better.

**Comment by vangent** [comment]
@ijt , I am waiting for your review since I added you as the `Assignee`.

**Comment by ijt** [comment]
Looking.

File: internal/testing/setup/setup.go [hunk 1]
```diff
@@ -120,13 +121,23 @@ func NewGCPgRPCConn(ctx context.Context, t *testing.T, endPoint string) (*grpc.C
...
 
 	opts, done := replay.NewGCPDialOptions(t, mode, t.Name()+".replay")
-	opts = append(opts, grpc.WithTransportCredentials(grpccreds.NewClientTLSFromCert(nil, "")))
 	if mode == recorder.ModeRecording {
+		// Add credentials for real RPCs.
 		creds, err := gcp.DefaultCredentials(ctx)
 		if err != nil {
 			t.Fatal(err)
 		}
+		opts = append(opts, grpc.WithTransportCredentials(grpccreds.NewClientTLSFromCert(nil, "")))
```
Comment (ijt) [inline internal/testing/setup/setup.go]: This seems unrelated.

Comment (ijt) [reply internal/testing/setup/setup.go]: Okay, makes sense.

**Pull request merged**